python/llm/example/CPU/HF-Transformers-AutoModels/Model/internlm-xcomposer/README.md

```diff
@@ -0,0 +1,91 @@
+
+```
+
+### 2. Download Model and Replace File
+If you select the InternLM_XComposer model ([internlm/internlm-xcomposer-vl-7b](https://huggingface.co/internlm/internlm-xcomposer-vl-7b)), please note that their code (`modeling_InternLM_XComposer.py`) does not support inference on CPU. To address this issue, we have provided the updated file ([internlm-xcomposer-vl-7b/modeling_InternLM_XComposer.py](./internlm-xcomposer-vl-7b/modeling_InternLM_XComposer.py), which can be used to conduct inference on CPU.
+
+#### 2.1 Download Model
+You could use the following code to download [internlm/internlm-xcomposer-vl-7b](https://huggingface.co/internlm/internlm-xcomposer-vl-7b) with a specific snapshot id. Please note that the `modeling_InternLM_XComposer.py` file that we provide are based on these specific commits.
+
+```
+from huggingface_hub import snapshot_download
+
+# for internlm/internlm-xcomposer-vl-7b
+model_path = snapshot_download(repo_id='internlm/internlm-xcomposer-vl-7b',
+                               revision="b06eb0c11653fe1568b6c5614b6b7be407ef8660",
+                               cache_dir="dir/path/where/model/files/are/downloaded")
+print(f'internlm/internlm-xcomposer-vl-7b checkpoint is downloaded to {model_path}')
+```
+
+#### 2.2 Replace `modeling_InternLM_XComposer.py`
+For `internlm/internlm-xcomposer-vl-7b`, you should replace the `modeling_InternLM_XComposer.py` with [internlm-xcomposer-vl-7b/modeling_InternLM_XComposer.py](./internlm-xcomposer-vl-7b/modeling_InternLM_XComposer.py).
+
+
+### 3. Run
+After setting up the Python environment, you could run the example by following steps.
+
+> **Note**: When loading the model in 4-bit, BigDL-LLM converts linear layers in the model into INT4 format. In theory, a *X*B model saved in 16-bit will requires approximately 2*X* GB of memory for loading, and ~0.5*X* GB memory for further inference.
+>
+> Please select the appropriate size of the LLaVA model based on the capabilities of your machine.
+
+#### 3.1 Client
+On client Windows machines, it is recommended to run directly with full utilization of all cores:
+```powershell
+python ./chat.py --image-path aiyinsitan.jpg
+```
+More information about arguments can be found in [Arguments Info](#33-arguments-info) section. The expected output can be found in [Sample Output](#34-sample-output) section.
+
+#### 3.2 Server
+For optimal performance on server, it is recommended to set several environment variables (refer to [here](../README.md#best-known-configuration-on-linux) for more information), and run the example with all the physical cores of a single socket.
+
+E.g. on Linux,
+```bash
+# set BigDL-Nano env variables
+source bigdl-nano-init
+
+# e.g. for a server with 48 cores per socket
+export OMP_NUM_THREADS=48
+numactl -C 0-47 -m 0 python ./chat.py --image-path aiyinsitan.jpg
```

Comment:
example still uses aiyinstan.jpg. 

You can change to demo.jpg. And then add a name to the reference picture. 
```
numactl -C 0-47 -m 0 python ./chat.py --image-path demo.jpg
```